test/BinTestCases.test.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+/* globals describe it */
+const should = require("should");
+const path = require("path");
+const fs = require("fs");
+const childProcess = require("child_process");
+
+describe("BinTestCases", () => {
+	const casesPath = path.join(__dirname, "binCases");
+	let categories = fs.readdirSync(casesPath);
+	categories = categories.map(cat => {
+		return {
+			name: cat,
+			tests: fs.readdirSync(path.join(casesPath, cat)).filter(function(folder) {
+				return folder.indexOf("_") < 0;
+			})
+		};
+	});
+
+	categories.forEach(category => {
+		describe(category.name, () => {
+			category.tests.forEach(testName => {
+				let suite = describe(testName, () => {});
+				it(`${testName} should compile from bin`, function(done) {
+					this.timeout(30000);
+					const testDirectory = path.join(casesPath, category.name, testName);
+					const outputDirectory = path.join(__dirname, "js", "bin", category.name, testName);
+					const configPath = path.join(testDirectory, "webpack.config.js");
+					const outputFilename = "[name].chunk.js";
+
+					function execHandler(error, stdout, stderr) {
+						if(stderr)
+							throw new Error(error);
+						if(error !== null)
+							throw new Error(error);
+
+						stdout.should.be.ok;
+						stderr.should.be.empty;
+						done();
+					}
+
+					childProcess.exec(`node ./bin/webpack.js --config ${configPath} --output-filename ${outputFilename} --output-path ${outputDirectory}`, execHandler);
```

Comment:
It would be great to have an `ARGUMENTS` file in the test directory which is used. This file may include placeholders to the OUTPUT_DIRECTORY. `webpack` should be invoke with working directory = test directory. After execution the result (exit code, stdout, stderr) should be passed to a validation function placed inside of the test directory (i. e. `test.js`: `module.exports = function(code, stdout, stderr, outputDirectory)`).